File: MetricsCalculators/test/cpp/Transaction.cpp
#include "Transaction.h"

Transaction::Transaction(unsigned int clientId,  Date & date, vector<string> products){
	this->clientId = clientId;
	this->date = date;
	this->products = products;
}

unsigned int Transaction::getClientId() const{
  return clientId;
}

vector<string> Transaction::getProducts() const{
	return products;
}

Date& Transaction::getDate(){
	return date;
}

bool Transaction::compareTo(const Transaction & trans){
	return this->date.compareTo(trans.date);
}

string Transaction::toString(){
	ostringstream oss;
	oss << clientId << " ; " << date.toString() << " ; ";
	
	for (unsigned int j = 0; j < products.size(); j++)
	{
		oss << products[j];
		if (j != products.size() - 1)
			oss << ", ";
	}
	oss << "\n";
		
	return oss.str();;
}

File: MetricsCalculators/test/java/src/main/java/store/Transaction.java
package store;

import java.util.Vector;

public class Transaction {
	int clientId;
	Date date; // DD/MM/AAAAA

	Vector<String> products;

	public Transaction(int clientId, Date date, Vector<String> products) {
		this.clientId = clientId;
		this.date = date;
		this.products = products;
	}

	public int getClientId() {
		return clientId;
	}

	public Vector<String> getProducts() {
		return products;
	}

	public Date getDate() {
		return date;
	}

	public boolean compareTo(Transaction transaction) {
		return date.compareTo(transaction.date);
	}

	public String toString() {
		
		String out = clientId + " ; " + date.toString() + " ; ";
		
		for (int j = 0; j < products.size(); j++) {
			out += products.elementAt(j);
			if (j != products.size() - 1)
				out += ", ";
		}
		out += "\n";
		
		return out;
	}

}
